internal/service/alarms/api/openapi.yaml

```diff
@@ -0,0 +1,756 @@
+
+externalDocs:
+  description: O-RAN apis from O-RAN.WG6.O2IMS-INTERFACE-R003-v06.00 (June 2024)
+  url: https://specifications.o-ran.org/download?id=674
+
+servers:
+  - url: http://localhost:8080
+    description: O-RAN Alarms Server
+
+tags:
+  - name: alarms
+    description: Alarm management
+  - name: subscriptions
+    description: Alarm subscription management
+  - name: probableCauses
+    description: Probable cause information
+  - name: internal
+    description: |
+      Internal endpoints. This is tag is also useful generating client code.
+
+paths:
+  /O2ims_infrastructureMonitoring/v1/alarms:
+    get:
+      operationId: GetAlarms
+      summary: Retrieve the list of alarms
+      tags:
+        - alarms
+      parameters:
+        - in: query
+          name: filter
+          schema:
+            type: string
+          allowReserved: true
+          description: Optional filter parameter. Only the attributes in AlarmEventRecord are filterable.
+          example: (eq,perceivedSeverity,0)
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/AlarmEventRecord'
+        '400':
+          description: Bad request
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+        '403':
+          description: Forbidden.
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+        '404':
+          description: Content not found.
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+        '500':
+          description: Internal server error.
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+
+  /O2ims_infrastructureMonitoring/v1/alarms/{alarmEventRecordId}:
+    get:
+      operationId: GetAlarm
+      summary: Retrieve exactly one alarm
+      tags:
+        - alarms
+      parameters:
+        - in: path
+          name: alarmEventRecordId
+          required: true
+          schema:
+            type: string
+            format: uuid
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/AlarmEventRecord'
+        '404':
+          description: The specified AlarmEventRecord was not found.
+          content:
+            application/problem+json:
+              schema:
+                $ref: '#/components/schemas/ProblemDetails'
+
+    patch:
+      operationId: AckAlarm
+      summary: Modify exactly one alarm to acknowledge
```

Comment:
to acknowledge and clear

Reply:
I think decided that we dont support clearing (diverging from the spec). This endpoint will simply take the `alarmAcknowledged` for patch request. Do we think we should follow the spec but in code we just consider any other value as no-op?

Reply:
I see. Didn't know only ack was going to be supported